lib/JITServer/JITServer.cpp

```diff
@@ -131,7 +131,13 @@ ServerInitializeThreadContext(
 {
     AUTO_NESTED_HANDLED_EXCEPTION_TYPE(static_cast<ExceptionType>(ExceptionType_OutOfMemory | ExceptionType_StackOverflow));
 
-    ServerThreadContext * contextInfo = HeapNew(ServerThreadContext, threadContextData);
+    ServerThreadContext * contextInfo = HeapNewNoThrow(ServerThreadContext, threadContextData);
```

Comment:
btw, looks we don't need AUTO_NESTED_HANDLED_EXCEPTION_TYPE anywhere in this file, that looks like mostly for run time in script to prevent unknown exception get out of scope. the Assertion inside is always true here in this file (see !JsUtil::ExternalApi::IsScriptActiveOnCurrentThreadContext() )

---
In reply to: [81828161](https://github.com/Microsoft/ChakraCore/pull/1689#discussion_r81828161) [](ancestors = 81828161,81825163)